utilities/PerformanceTesting/GetAverages.py

```diff
@@ -0,0 +1,74 @@
+import sys, argparse, json
+
+def getFilePaths(file_prefix, num_files):
+    files = []
+
+    for i in range(num_files):
+        files.append(f"{file_prefix}{i}.json")
+
+    return files
+
+def getJsons(file_paths):
+    jsons = []
+
+    for file_path in file_paths:
+        f = open(file_path)
+        j = json.load(f)
+        jsons.append(j)
+
+    return jsons
+
+def isFloat(s):
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
+
+def findAverageForMetric(jsons, key, idx):
+    num_files = len(jsons)
+    total = 0
+    count = 0
+
+    for j in jsons:
+        if j[key][idx]['name'] == jsons[0][key][idx]['name']:
```

Comment:
If they have a name, can we simply by not having this as a list? i.e.

```
{
    'key': {
        'name': value,
        'name2': value,
        # ... etc
    }
}
```

Because then we won't need to make these checks, we won't need to iterate over all the objects in a list, we just look up the name that we want to process and accumulate the average. The logic as it currently stands is quite complex to follow, with a fair few edge cases, some of which aren't handled currently

Reply:
I am not sure if this is a good idea. We might have additional fields such as standard deviation after running the GetAverages script. Eg

`{
     'key1': {[
          {
               "name": "metric1"
               "value": "123"
               "standard_deviation": "56"
          },
          {
               "name": "metric2"
               "value": "567"
               "standard_deviation": "12"
          }
     ]}
}`

Also name oftentimes has spaces and special characters, so I would prefer not to use it as a key.